codersdk/users.go

```diff
@@ -45,6 +45,7 @@ type User struct {
 	OrganizationIDs []uuid.UUID `json:"organization_ids" format:"uuid"`
 	Roles           []Role      `json:"roles"`
 	AvatarURL       string      `json:"avatar_url" format:"uri"`
+	LoginType       LoginType   `json:"login_type"`
```

Comment:
I am adding `LoginType` to the /users struct. 

This issue is being resolved soon with some open PRs I have: https://github.com/coder/coder/issues/5002

With this change, only admins can read users, so exposing this field is ok.
This solves some UI issues with this PR on not allowing role changing, and showing login type here: https://github.com/coder/coder/issues/8346